### DISCUSSION TITLE
Points of interest layers

### DISCUSSION THREAD

**Discussion by Ciutnik** [devemux86/cruiser#620] [Q&A] [answered]
What I do at the moment is that, in order to see the POIs I am interested in while browsing the map, I use Overpass-turbo. Unfortunately, the resulting sets of Favourites are sometimes several thousand points long! This obviously affects the performance of the map terribly.

But since we already have the same POI collection as part of the mapset, would it be possible to create a dynamic overlay displaying selected POI categories, for the current map view?

It's something like the POI search we already have, but constantly active, without the need to manually start it.

**Comment by devemux86** [comment]
@Ciutnik thanks for the suggestion.

We have 2 offline sources of map data:
- **Map files**: optimized for rendering, they contain the name and rendering tags.

- **POI files**: suitable for database search, they contain all OpenStreetMap tags.
  - We already use them in offline search and nearby POI in navigation.

What you describe is to use the POI files as layers?
Like we already use the favorites and their groups.

Tap to see the POI info, long press for their menu.
Which are already available when we do offline search.

Performance is still important, because of the background search in scroll / zoom.

**Comment by MotoUKRider** [comment]
After some private correspondence with @devemux86 , I'll throw my hat into the ring on this topic.

With the existing Map Layers (which currently show embedded/static POI icons) not allowing you to directly interact with them, we find ourselves in the amusing position of having to search for something (in the POI DB) that we can already see(!) A faster workflow would be the ability to browse the map, touch a POI icon (like a campsite) and have the option of doing something with it, like visit their website or phone them.

The POI Database as a set of icons we can turn on/off and directly access (for navigation/contact-details etc) would also follow behaviour of many other navigation solutions and almost certainly be a winner with _Stupid Motorcyclists_ :tm:

**Reply by 0709wiwiwi** [reply]
@MotoUKRider 
Apparently a feature as Locus map 4 offers with the Lopoints?  Fortunately I can disable this in the Content Panel. 
Opinions may differ, I found the numerous poi pop-ups when touching and scrolling through the map particularly annoying.

**Reply by MotoUKRider** [reply]
@0709wiwiwi Sausage-fingers?!

I can browse a Favorite-heavy map in Cruiser without triggering the popup balloons/context-menu, but can see how it may become an issue. Performing a quick test, if you _touch+move_ rather than _touch+hold_, there is no action.  I also experience this with other Apps.

**Reply by 0709wiwiwi** [reply]
No, no, it's different. The Lopoints function works well, but the density of (too) many poi is distracting. If every bench, planter, and other clutter is clickable, it's more of a hindrance than a help. Being able to disable that functionality is useful.

**Reply by MotoUKRider** [reply]
@0709wiwiwi That's an interesting point - I think bona-fide "static" objects like benches, gates and other street furniture should be excluded. Only `amenity` , `shop` , `tourism` etc  I suppose you have to include landmarks/viewpoints?  I would guess the criteria is would you want to navigate to a bollard or water-fountain?!

Some of the furniture was backed-off (or relegated to the "Misc" Layer) in the themes, anything like that which bothers you (too visible or too hidden) can be discussed if you ever spot a good/bad example.

**Reply by devemux86** [reply]
Indeed, such layers are mostly suitable for POIs that have useful OSM tags from which we can read their information.

**Comment by devemux86** [comment, marked as answer]
Points of interest layers is available in [**Cruiser 5.9.13**](https://github.com/devemux86/cruiser/discussions/901#discussioncomment-17849230).

Search for points of interest, filter and select categories to display with the search button.

Now you do not need to load external GPX files, e.g. camp sites, everything is included offline in Cruiser.

<img src="https://github.com/user-attachments/assets/554b3689-3ec0-42a8-b789-7a883e7dd3a7" width="50%"/><img src="https://github.com/user-attachments/assets/758ea43a-5f57-42cc-a818-c417c984816b" width="50%"/>

**Reply by devemux86** [reply]
Remember to update Mapsforge POI to the new, faster version with the menu "**Download POI**" on the local maps screen:
- https://github.com/devemux86/cruiser/discussions/930

**Reply by Ciutnik** [reply]
Hi. Although my personal situation has changed so much that I don't have the opportunity to use the Cruiser (apart from occasional hikes), I still love it and watch it evolve.

POI layers are a long-awaited functionality, and I thank you @devemux86 very much for implementing it 👍 
However, if I could have any wish, I would like to see the option to remember the set of selected POIs after restarting the application (@jcz7021 already mentioned this in another thread). And the second thing is the automatic refreshing of the layer after moving/zooming the map.

Thank you again and I assure you that I am here and I am still loyal to Cruiser ❤️

**Reply by WolfgangZ42** [reply]
All the best for you, @Ciutnik❗

**Reply by devemux86** [reply]
Remembering the selected POI after restarting the app can become complicated
when users activate too many and heavy layers resulting in the app becoming slow.
e.g. restaurants for the whole of Germany together with other heavy layers

So we need to think carefully about this.

**Reply by jcz7021** [reply]
> Remembering the selected POI after restarting the app can become complicated when users activate too many and heavy layers resulting in the app becoming slow. e.g. restaurants for the whole of Germany together with other heavy layers
> 
> So we need to think carefully about this.

I don't know how the feature was thinked at first.

As for now (directly displaying results on visible portion of the map and not refreshing when panning or zooming) it seems to operate like the "Search button" -> POI from my point of view.

I'm using cruiser totally offline, so for now when we compare the two features, they seems to be close in terms of features

| Feature | "Search button" -> POI | Map ->POI |
|---------|---------------------------|--------------|
| **Search by text** | ✅                | ✅              |
| **Search by text for onr or more specific categories** | ❌ | ✅ |
| **Search by category only (without text)** | ✅ | ✅ |
| **Search by multiple categories only (without text)** | ❌ | ✅ |
| **Search by already predifined category** | ✅<br>Bars search for amenity Pub or Bar (and maybe more) | ❌<br>Needs to select Foods - Pubs + Foods - Bars |

The difference is that with **"Search button" -> POI** we can preview results, pick on and decide to display all result on the map.

**Map -> POI**
I don't know if grouping is easy to do (built-in component) to organize POI with nested levels:
<img width="294" height="212" alt="image" src="https://github.com/user-attachments/assets/af60ecce-5d04-4a33-bccb-c6a82dec6b18" />

If we keep as it is for now (not refreshing while panning), just persisting the checked filters only (not restoring the POI visually on the map) could be ok right ?

**Reply by Ciutnik** [reply]
Is it possible for this new feature to also find polygons? I see that they (or at least some of them) are included in the POIs and are findable in classic search.

--- 

OT: I'm currently watching a terrifying report on the news about a firefight near Athens. There's also a report about a collision between firefighting helicopters. Truly terrifying. The situation is similar in Spain and France.

**Reply by devemux86** [reply]
> Is it possible for this new feature to also find polygons? I see that they (or at least some of them) are included in the POIs and are findable in classic search.

Points of interest are of both types:
- OSM points
- OSM polygons at their centroids

**Reply by devemux86** [reply]
The POI layers are just the first implementation, they will be improved in the next updates.

As for the search button and the new POI layers,
they will probably be merged into one screen to avoid the duplicate user interface.

**Reply by Ciutnik** [reply]
Perhaps it would be nice to be able to search (and showing in the panel) for selected POI categories along the route, just like it is now with Favorites.

**Reply by devemux86** [reply]
> Perhaps it would be nice to be able to search (and showing in the panel) for selected POI categories along the route, just like it is now with Favorites.

Points of interest can already be searched and displayed along the route, as well as in navigation.

Specifically in navigation, only certain categories of points of interest have real value for display their panels,
i.e. points of interest that are close to the road network.

You can suggest categories of points of interest for discussion to include them in the navigation panels.
